File: app/src/main/java/com/ahmettekin/retrofitjava/model/WeatherModel.java
package com.ahmettekin.retrofitjava.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class WeatherModel {

    @SerializedName("list")
    public ArrayList<WeatherItem> weatherItems;

    public ArrayList<WeatherItem> getWeatherItems() {
        return weatherItems;
    }

    public void setWeatherItems(ArrayList<WeatherItem> weatherItems) {
        this.weatherItems = weatherItems;
    }

    public class WeatherItem{
        private Weather[] weather;

        private Main main;

        public Weather[] getWeather() {
            return weather;
        }

        public void setWeather(Weather[] weather) {
            this.weather = weather;
        }

        public Main getMain() {
            return main;
        }

        public void setMain(Main main) {
            this.main = main;
        }

        public class Weather {
            private long id;
            private String main;
            private String description;
            private String icon;

            public long getId() {
                return id;
            }

            public void setId(long id) {
                this.id = id;
            }

            public String getMain() {
                return main;
            }

            public void setMain(String main) {
                this.main = main;
            }

            public String getDescription() {
                return description;
            }

            public void setDescription(String description) {
                this.description = description;
            }

            public String getIcon() {
                return icon;
            }

            public void setIcon(String icon) {
                this.icon = icon;
            }
        }

        public class Main {

            private double temp;
            @SerializedName("feels_like")
            private double feelsLike;
            @SerializedName("temp_min")
            private double tempMin;
            @SerializedName("temp_max")
            private double tempMax;
            private double pressure;
            private double humidity;

            public double getTemp() {
                return temp;
            }

            public void setTemp(double temp) {
                this.temp = temp;
            }

            public double getFeelsLike() {
                return feelsLike;
            }

            public void setFeelsLike(double feelsLike) {
                this.feelsLike = feelsLike;
            }

            public double getTempMin() {
                return tempMin;
            }

            public void setTempMin(double tempMin) {
                this.tempMin = tempMin;
            }

            public double getTempMax() {
                return tempMax;
            }

            public void setTempMax(double tempMax) {
                this.tempMax = tempMax;
            }

            public double getPressure() {
                return pressure;
            }

            public void setPressure(double pressure) {
                this.pressure = pressure;
            }

            public double getHumidity() {
                return humidity;
            }

            public void setHumidity(double humidity) {
                this.humidity = humidity;
            }
        }

    }



}

File: app/src/main/java/com/ahmettekin/weatherappkotlin/model/WeatherModel.kt
package com.ahmettekin.weatherappkotlin.model


import com.google.gson.annotations.SerializedName

class WeatherModel(
    @SerializedName("cnt")
    var cnt: Int,
    @SerializedName("list")
    var list: List<WeatherItem>
) {
    class WeatherItem(
        @SerializedName("clouds")
        var clouds: Clouds,
        @SerializedName("coord")
        var coord: Coord,
        @SerializedName("dt")
        var dt: Int,
        @SerializedName("id")
        var id: Int,
        @SerializedName("main")
        var main: Main,
        @SerializedName("name")
        var name: String,
        @SerializedName("sys")
        var sys: Sys,
        @SerializedName("visibility")
        var visibility: Int,
        @SerializedName("weather")
        var weather: List<Weather>,
        @SerializedName("wind")
        var wind: Wind
    ) {
        class Clouds(
            @SerializedName("all")
            var all: Int
        )

        class Coord(
            @SerializedName("lat")
            var lat: Double,
            @SerializedName("lon")
            var lon: Double
        )

        class Main(
            @SerializedName("feels_like")
            var feelsLike: Double,
            @SerializedName("humidity")
            var humidity: Int,
            @SerializedName("pressure")
            var pressure: Int,
            @SerializedName("temp")
            var temp: Double,
            @SerializedName("temp_max")
            var tempMax: Double,
            @SerializedName("temp_min")
            var tempMin: Double
        )

        class Sys(
            @SerializedName("country")
            var country: String,
            @SerializedName("sunrise")
            var sunrise: Int,
            @SerializedName("sunset")
            var sunset: Int,
            @SerializedName("timezone")
            var timezone: Int
        )

        class Weather(
            @SerializedName("description")
            var description: String,
            @SerializedName("icon")
            var icon: String,
            @SerializedName("id")
            var id: Int,
            @SerializedName("main")
            var main: String
        )

        class Wind(
            @SerializedName("deg")
            var deg: Int,
            @SerializedName("speed")
            var speed: Double
        )
    }
}